alicloud/service_alicloud_sag.go

```diff
@@ -0,0 +1,551 @@
+package alicloud
+
+import (
+	"strings"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/smartag"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+type SagService struct {
+	client *connectivity.AliyunClient
+}
+
+// Flattens an array of SagInstances into a []map[string]string
+func (s *SagService) FlattenSagInstancesMappings(list []smartag.SmartAccessGateway) []map[string]interface{} {
```

Comment:
I think the method `FlattenSagInstancesMappings ` has never been used. Please remove it.